Sources/MessagingPush/MessagingPush.swift

```diff
@@ -66,7 +66,7 @@ public class MessagingPush: ModuleTopLevelObject<MessagingPushInstance>, Messagi
         let logger = diGraph.logger
         logger.debug("Setting up MessagingPush module...")
 
-        if diGraph.sdkConfig.autoTrackPushEvents {
+        if diGraph.sdkConfig.autoTrackPushMetricEvents.isOpenedEnabled {
             diGraph.automaticPushClickHandling.start()
         }
```

Comment:
This is a blocker comment, because it could cause unexpected behaviors for customers. 

The automatic push click handling feature does much more then track opened metrics. It also: 
1. Decides [if a push sent from CIO should be shown while app in foreground](https://github.com/customerio/customerio-ios/blob/873eb43f111a4b451ed271b79e8a101d520ee98d/Sources/MessagingPush/PushHandling/iOSPushEventListener.swift#L55). 
2. Opens a deep link if one is attached to a push and a push is clicked. 

If a customer disables opened metrics tracking, they should be able to expect these other features continue to work.